src/Datadog.NativeLoader/pal.cpp

```diff
@@ -0,0 +1,43 @@
+#include "pal.h"
+
+#if _WIN32
+#include <Windows.h>
+#else
+#include "dlfcn.h"
+#endif
+
+#include "logging.h"
+
+namespace datadog
+{
+namespace nativeloader
+{
+
+#if _WIN32
+    void* LoadDynamicLibrary(std::string filePath)
+    {
+        Debug("LoadLibrary: ", filePath);
+        return LoadLibrary(ToWSTRING(filePath).c_str());
+    }
+
+    void* GetExternalFunction(void* instance, std::string funcName)
+    {
+        Debug("GetExternalFunction: ", funcName);
+        return (void*)GetProcAddress(instance, funcName.c_str());
+    }
+#else
+    void* LoadDynamicLibrary(std::string filePath)
+    {
+        Debug("LoadLibrary: ", filePath);
+        return dlopen(filePath.c_str(), RTLD_LOCAL | RTLD_LAZY);
```

Comment:
If it fails to load the library, we should log it (linux dlerror)

Reply:
done in 35452f5